horizons/main.py

```diff
@@ -445,36 +445,45 @@ def _load_cmd_map(savegame, ai_players, force_player_id=None):
 	return True
 
 def _find_matching_map(name_or_path, savegames):
-	"""*name_or_path* is either a map/savegame name or path to a map/savegame file."""
+	"""Find a map by name or path specifiec by user.
+	@param name_or_path: either a map/savegame name or path to a map/savegame file
+	@return: str with the path to the map/savegame file on success"""
 	game_language = horizons.globals.fife.get_locale()
-	# now we have "_en.yaml" which is set to language_extension variable
-	language_extension = '_' + game_language + '.' + SavegameManager.scenario_extension
-	map_file = None
-	for filename, name in zip(*savegames):
-		if name in (name_or_path, name_or_path + language_extension):
-			# exact match or "tutorial" matching "tutorial_en.yaml"
-			return filename
-		if name.startswith(name_or_path): # check for partial match
-			if map_file is not None:
-				# multiple matches, collect all for output
-				map_file += '\n' + filename
-			else:
-				map_file = filename
-	if map_file is not None:
-		if len(map_file.splitlines()) > 1:
-			print("Error: Found multiple matches:")
-			for name_or_path in map_file.splitlines():
-				print(os.path.basename(name_or_path))
-			return
-		else:
-			return map_file
-	else: # not a savegame, check for path to file or fail
-		if os.path.exists(name_or_path):
+
+	# Check if name_or_path is a path
```

Comment:
This comment is redundant and should be removed. `os.path.exists` doesn't need any explanation IMO.